hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/compact/CompactStreamSerializer.java

```diff
@@ -259,7 +260,7 @@ private CompactSerializableRegistration getOrCreateRegistration(String typeName)
                 return null;
             }
             try {
-                Object object = ClassLoaderUtil.newInstance(clazz.getClassLoader(), clazz);
+                Object object = newInstanceBypassingConstructor(clazz);
```

Comment:
If I understand this correctly then it skips the constructor in all cases, not only if it's not available, correct? Are we sure we want to do it this way? Maybe we can fallback to bypassing constructor only if the default constructor doesn't exist?

Reply:
Hi, yes, it skips constructor in all cases. I thought it's the easiest way, but can change it too - no preferences here from my side